### PR TITLE
Add the cuSpatial UDF case

### PR DESCRIPTION
**Pull request by firestarman**
It includes 4 major components.

1. the UDF implementation under src folder (Java and C++)
2. the main app code in file spatial_join.py
3. a script file to run this example against a Spark standalone cluster
4. tools to convert the points data from cny to parquet for Spark (requiring cuDF python package installed)

Signed-off-by: Liangcai Li <liangcail@nvidia.com>

### CODE REVIEW

**Comment by nvliyuan** [comment]
can you please target this pr to v22.04? since rapids plugin is really close to release, we may need some time to review this pr.

**Comment by GaryShen2008** [comment]
Added notebooks and sample data.
updated README for running on AWS Databricks.

**Comment by GaryShen2008** [comment]
We should add another Dockerfile for building.

File: examples/cuSpatial-UDF/notebooks/spatial_join_standalone.ipynb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+   "id": "cce700a8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# prepare shape files\n",
+    "spark.sparkContext.addFile(\"/data/cuspatial/taxi_zones/shape/taxi_zones.shp\")\n",
+    "spark.sparkContext.addFile(\"/data/cuspatial/taxi_zones/shape/taxi_zones.shx\")\n",
+    "\n",
+    "spark.sparkContext.addFile(\"/data/cuspatial/taxi_zones/shape/nyct2000.shp\")\n",
+    "spark.sparkContext.addFile(\"/data/cuspatial/taxi_zones/shape/nyct2000.shx\")\n",
```
Comment (GaryShen2008) [inline examples/cuSpatial-UDF/notebooks/spatial_join_standalone.ipynb:39]: Can we use toy dataset?
And why set two shapefiles?

Comment (firestarman) [reply examples/cuSpatial-UDF/notebooks/spatial_join_standalone.ipynb]: removed

**Review by GaryShen2008 — APPROVED** [review]

**Pull request merged**